SU2_CFD/src/iteration_structure.cpp

```diff
@@ -1285,51 +1285,34 @@ void CFEAIteration::Iterate(COutput *output,
 
     /*--- THIS IS THE INCREMENTAL LOAD APPROACH (only makes sense for nonlinear) ---*/
 
-    /*--- Set the initial condition: store the current solution as Solution_Old ---*/
-
-    feaSolver->SetInitialCondition(geometry[val_iZone][val_iInst],
-                                   solver[val_iZone][val_iInst], config[val_iZone], TimeIter);
-
     /*--- Assume the initial load increment as 1.0 ---*/
 
-    loadIncrement = 1.0;
-    feaSolver->SetLoad_Increment(loadIncrement);
-    feaSolver->SetForceCoeff(loadIncrement);
+    feaSolver->SetLoad_Increment(0, 1.0);
+    feaSolver->SetForceCoeff(1.0);
 
     /*--- Run two nonlinear iterations to check if incremental loading can be skipped ---*/
 
     for (IntIter = 0; IntIter < 2; ++IntIter) {
 
       config[val_iZone]->SetInnerIter(IntIter);
 
-      /*--- Run the first iteration ---*/
-
       feaIntegration->Structural_Iteration(geometry, solver, numerics, config,
                                            RUNTIME_FEA_SYS, val_iZone, val_iInst);
 
-      /*--- Write the convergence history (first computes Von Mises stress) ---*/
-
-      Monitor(output, integration, geometry, solver, numerics, config,
-              surface_movement, grid_movement, FFDBox, val_iZone, INST_0);
+      StopCalc = Monitor(output, integration, geometry, solver, numerics, config,
+                         surface_movement, grid_movement, FFDBox, val_iZone, INST_0);
     }
 
-    bool meetCriteria;
-    su2double Residual_UTOL, Residual_RTOL, Residual_ETOL;
-    su2double Criteria_UTOL, Criteria_RTOL, Criteria_ETOL;
+    /*--- Early return if we already meet the convergence criteria. ---*/
+    if (StopCalc) return;
 
-    Criteria_UTOL = config[val_iZone]->GetIncLoad_Criteria(0);
-    Criteria_RTOL = config[val_iZone]->GetIncLoad_Criteria(1);
-    Criteria_ETOL = config[val_iZone]->GetIncLoad_Criteria(2);
+    /*--- Check user-defined criteria to either increment loads or continue with NR iterations. ---*/
 
-    Residual_UTOL = log10(feaSolver->LinSysSol.norm());
-    Residual_RTOL = log10(feaSolver->LinSysRes.norm());
-    Residual_ETOL = log10(feaSolver->LinSysSol.dot(feaSolver->LinSysRes));
+    bool meetCriteria = true;
+    for (int i = 0; i < 3; ++i)
```

Comment:
can this 3 be an NDIM or other named var?

Reply:
No, it is the number of convergence criteria for nonlinear FEM there are 3 even for 2D problems.

Reply:
Sounds good. I think you get my gist.. just looking for places where enums/named vars can be used instead of literals for clarity